providers/xvasynth.py

```diff
@@ -0,0 +1,124 @@
+            )
+        if not config.game_folder_name:
+            errors.append(
+                "Missing game folder name where xvasynth voice is located, e.g., masseffect, skyrim, etc. Please provide the game folder name to use xvasynth voice."
+            )
+        if not config.voice:
+            errors.append(
+                "Missing voice name.  Please provide voice name to use xvasynth."
+            )
+            
+        self.xvasynth_path = config.xvasynth_path
+        self.process_device = config.process_device
+        self.times_checked_xvasynth = 0
+        # check if xvasynth is running a few times, if cannot find it, send error
+        while self.times_checked_xvasynth < 5:
+            check = self.check_if_xvasynth_is_running()
+            if check == "ok":
+                break
+            self.times_checked_xvasynth+=1
+        
+        if check != "ok":
+            errors.append(check)
+        # if xvasynth is found, load initial XVASynth voice
+        else:
+            load_ok = self.load_xvasynth_model(path_to_xvasynth=config.xvasynth_path, game_folder=config.game_folder_name, voice=config.voice, language=config.language)
+            if load_ok != "ok":
+                errors.append(load_ok)
+            self.current_voice = config.voice
+
+        return errors
+
+    def play_audio(
+        self, text: str, config: XVASynthTtsConfig, sound_config: SoundConfig
+    ):
+        load_ok = self.load_xvasynth_model(path_to_xvasynth=config.xvasynth_path, game_folder=config.game_folder_name, voice=config.voice, language=config.language)
+        if load_ok != "ok":
+            print("There was a problem loading your XVASynth voice, check your voice name and game folder for your voice in XVAsynth")
+        voiceline = text
+        file_dir = path.abspath(path.dirname(__file__))
+        wingman_dir = path.abspath(path.dirname(file_dir))
+        final_voiceline_file =  wingman_dir + "\\audio_output\\xvasynth.wav"
+
+        if path.exists(final_voiceline_file):
+            os.remove(final_voiceline_file)
+
+        # Synthesize voiceline
+        synthesize_url= XVASynthModel.SYNTHESIZE_URL
+        data = {
+            'pluginsContext': '{}',
+            'modelType': 'xVAPitch',
+            'sequence': voiceline,
+            'pace': config.pace,
+            'outfile': final_voiceline_file,
+            'vocoder': 'n/a',
+            'base_lang': config.language,
+            'base_emb': '[]',
+            'useSR': config.useSR,
+            'useCleanup': config.useCleanup,
+        }
+        response = requests.post(synthesize_url, json=data)
+        audio_player = AudioPlayer()
+        audio, sample_rate = audio_player.get_audio_from_file(final_voiceline_file)
+        audio_player.stream_with_effects(input_data=(audio, sample_rate), config=sound_config)
+
+    def check_if_xvasynth_is_running(self):
+        try:
+            response = requests.get('http://127.0.0.1:8008/')
```

Comment:
This shouldn't be hard-coded, right?

Reply:
So the interesting thing is that XVASynth was originally designed to be a standalone line generation program.  It happens to have a server but it isn’t exposed to the end user normally, it is used for its own communications.  I will double check but presently I don’t believe there is a way for the end user to modify this, so that’s why I used a constant in general but you’re right I should have the constant (if that approach is used) or variable (if that approach is used) in this piece of code.  Probably always best to expose the server URL as a variable to the user as you say even if we say something like “this value ordinarily should not be changed” just in case some advanced user figures out a way to reroute the XVASynth server somehow.  I’ll work on this by tomorrow morning Eastern and update the PR so it can be merged and tested.  Appreciate it!

Reply:
fixed